python/cugraph/structure/number_map.py

```diff
@@ -314,7 +315,7 @@ def compute(self):
 
                 numbering_map = rehashed_with_partition_id.map_partitions(
                     lambda df: df.groupby(
-                        self.col_names + ["hash", "partition"]
+                        self.col_names + ["hash", "partition"], sort=True
```

Comment:
This does not require `sort=True`.  The groupby is followed explicitly by a sort.